crates/red_knot/tests/cli.rs

```diff
@@ -244,6 +350,25 @@ fn unknown_rules() -> anyhow::Result<()> {
     Ok(())
 }
 
+/// Red Knot warns about unknown rules specified in a CLI argument
+#[test]
+fn cli_unknown_rules() -> anyhow::Result<()> {
```

Comment:
I played around with the CLI for a bit and one thing I tried initially was to silence this warning
```
error[lint:invalid-assignment] /home/shark/playground/test.py:4:1 Object of type `tuple[Literal["bar"]]` is not assignable to attribute `__slots__` of type `tuple[Literal["foo"]]`
```
by copying (what I thought was) the rule name, and then added `--ignore lint:invalid-assignment`.

This leads to
```
warning[unknown-rule] Unknown lint rule `lint:invalid-assignment`
```

It would be nice if we could show a better error message for this specific case (by checking if the rule exists after stripping `lint:` from the start of the argument?):

```
warning[unknown-rule] Unknown lint rule `lint:invalid-assignment`. Did you mean `invalid-assignment`?
```

Reply:
Nice find. I was pretty sure it's something I called out in the CLI proposal but it seems I didn't. 

A general concern around using `lint:<name>` in diagnostics is that it prevents copy-pasting the diagnostic code into a configuration or even a suppression comment. It's something that bothers me about clippy where the diagnostic uses `-` but `allow` and `deny` require `_`.